contrib/setup-z3.sh

```diff
@@ -18,10 +26,20 @@ if [ ! -d "$DEPS/z3" ]; then
     # a pthread related issue
     # compiling with --single-threaded helps, but isn't a real solution
     # see https://github.com/Z3Prover/z3/issues/4554
-    ./configure --staticlib --single-threaded
-    cd build
-    make -j$(nproc)
+    # ./configure --staticlib --single-threaded
+    cmake -S . -B build -DCMAKE_BUILD_TYPE=Release -DZ3_BUILD_LIBZ3_SHARED=Off -DZ3_BUILD_LIBZ3_MSVC_STATIC=On
+    cmake --build build -j$NUM_CORES
     cd $DIR
 else
     echo "$DEPS/z3 already exists. If you want to rebuild, please remove it manually."
 fi
+
+if [ -f $DEPS/z3/build/libz3.a ]; then
+    echo "It appears z3 was setup successfully into $DEPS/z3."
+    echo "You may now install it with make ./configure.sh --z3 && cd build && make"
+else
+    echo "Building z3 failed."
+    echo "You might be missing some dependencies."
+    echo "Please see their github page for installation instructions: https://github.com/z3/z3"
+    exit 1
+fi
```

Comment:
same question -- is this necessary? What is the current behavior?

Reply:
Copied from contrib/setup-cvc5.sh as well.